core/helpers/EEH_Line_Item.helper.php

```diff
@@ -1637,72 +1912,156 @@ public static function non_empty_line_item(EE_Line_Item $line_item)
     }
 
 
+    /**
+     * Cycles through all of the ticket line items for the supplied total line item
+     * and ensures that the line item's "is_taxable" field matches that of its corresponding ticket
+     *
+     * @param EE_Line_Item $total_line_item
+     * @since $VID:$
+     * @throws EE_Error
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
+     * @throws ReflectionException
+     */
+    public static function resetIsTaxableForTickets(EE_Line_Item $total_line_item)
+    {
+        $ticket_line_items = self::get_ticket_line_items($total_line_item);
+        foreach ($ticket_line_items as $ticket_line_item) {
+            if ($ticket_line_item instanceof EE_Line_Item
+                && $ticket_line_item->OBJ_type() === EEM_Line_Item::OBJ_TYPE_TICKET
+            ) {
+                $ticket = $ticket_line_item->ticket();
+                if ($ticket instanceof EE_Ticket && $ticket->taxable() !== $ticket_line_item->is_taxable()) {
+                    $ticket_line_item->set_is_taxable($ticket->taxable());
+                    $ticket_line_item->save();
+                }
+            }
+        }
+    }
+
+
 
     /**************************************** @DEPRECATED METHODS *************************************** */
     /**
      * @deprecated
      * @param EE_Line_Item $total_line_item
-     * @return \EE_Line_Item
-     * @throws \EE_Error
+     * @return EE_Line_Item
+     * @throws EE_Error
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidInterfaceException
+     * @throws ReflectionException
      */
     public static function get_items_subtotal(EE_Line_Item $total_line_item)
     {
-        EE_Error::doing_it_wrong('EEH_Line_Item::get_items_subtotal()', __('Method replaced with EEH_Line_Item::get_pre_tax_subtotal()', 'event_espresso'), '4.6.0');
+        EE_Error::doing_it_wrong(
+            'EEH_Line_Item::get_items_subtotal()',
+            esc_html__(
+                'Method replaced with EEH_Line_Item::get_pre_tax_subtotal()',
```

Comment:
In all these `doing_it_wrong` strings, we probably should not be translating the class and method name.  However, I understand if you decide not to make the change because it would break existing strings.

Reply:
no I think fixing is more important, plus, this allows eon translation to rule them all: `Method replaced with %1$s`